### PR TITLE
SentinelOne:Connector+parser

### PR DESCRIPTION
**Pull request by averbn**
Fixes #

## Proposed Changes

  - added Connector
  - added parser

### CODE REVIEW

**Review by v-admahe — CHANGES_REQUESTED** [review]

File: Parsers/SentinelOne/SentinelOne.txt [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+                MachineType,
+                MitigationMode,
+                MitigationModeSuspicious,
+                ModelName,
+                NetworkInterfaces,
+                NetworkQuarantineEnabled,
+                NetworkStatus,
+                OperationalState,
+                OsArch,
+                DvcOs,
```
Comment (v-admahe) [inline Parsers/SentinelOne/SentinelOne.txt:253]: Should it not be 'SrcDvcOs'? Please refer https://docs.microsoft.com/en-us/azure/sentinel/normalization-schema for field names.

Thanks,
Aditi Maheshwari

**Review by v-admahe — CHANGES_REQUESTED** [review]

File: Parsers/SentinelOne/SentinelOne.txt [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+                AccountName,
+                ActivityType,
+                EventCreationTime,
+                DataAccountName,
+                DataFullScopeDetails,
+                DataScopeLevel,
+                DataScopeName,
+                DataSiteId,
+                DataSiteName,
+                UserName,
```
Comment (v-admahe) [inline Parsers/SentinelOne/SentinelOne.txt:156]: Should it not be 'SrcUserName'? Please refer https://docs.microsoft.com/en-us/azure/sentinel/normalization-schema for field names.

Thanks,
Aditi Maheshwari

**Review by v-admahe — CHANGES_REQUESTED** [review]

**Comment by v-admahe** [comment]
@averbn : Could you please provide the sample data in json format. Please add the sample data file in PR as well.

Thanks,
Aditi Maheshwari

**Comment by averbn** [comment]
> @averbn : Could you please provide the sample data in json format. Please add the sample data file in PR as well.
> 
> Thanks,
> Aditi Maheshwari

Sure

File: Parsers/SentinelOne/SentinelOne.txt (hunk 2 above)
Comment (averbn) [reply Parsers/SentinelOne/SentinelOne.txt:156]: Fixed

File: Parsers/SentinelOne/SentinelOne.txt [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+                ActiveThreats,
+                AgentVersion,
+                AllowRemoteShell,
+                AppsVulnerabilityStatus,
+                ComputerName,
+                ConsoleMigrationStatus,
+                CoreCount,
+                CpuCount,
+                CpuId,
+                Domain,
```
[the remark this answers is not in the document]
Comment (averbn) [reply Parsers/SentinelOne/SentinelOne.txt:223]: Fixed

File: Parsers/SentinelOne/SentinelOne.txt [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+                LastIpToMgmt,
+                LastLoggedInUserName,
+                LicenseKey,
+                LocationEnabled,
+                LocationType,
+                Locations,
+                MachineType,
+                MitigationMode,
+                MitigationModeSuspicious,
+                ModelName,
```
[the remark this answers is not in the document]
Comment (averbn) [reply Parsers/SentinelOne/SentinelOne.txt:247]: Fixed

File: Parsers/SentinelOne/SentinelOne.txt (hunk 1 above)
Comment (averbn) [reply Parsers/SentinelOne/SentinelOne.txt:253]: Fixed

File: DataConnectors/SentinelOne/SentinelOne_API_FunctionApp.json [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+	],
+	"connectivityCriterias": [{
+			"type": "IsConnectedQuery",
+			"value": [
+				"SentinelOne_CL\n            | summarize LastLogReceived = max(TimeGenerated)\n            | project IsConnected = LastLogReceived > ago(30d)"
+			]
+		}
+	],
+	"availability": {
+		"status": 1,
```
[the remark this answers is not in the document]
Comment (averbn) [reply DataConnectors/SentinelOne/SentinelOne_API_FunctionApp.json:31]: Fixed

File: DataConnectors/SentinelOne/azuredeploy_Connector_SentinelOneAPI_AzureFunction.json [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+                        "APPLICATIONINSIGHTS_CONNECTION_STRING": "[reference(resourceId('microsoft.insights/components', variables('FunctionName')), '2015-05-01').ConnectionString]",
+                        "AzureWebJobsStorage": "[concat('DefaultEndpointsProtocol=https;AccountName=', toLower(variables('FunctionName')),';AccountKey=',listKeys(resourceId('Microsoft.Storage/storageAccounts', toLower(variables('FunctionName'))), '2019-06-01').keys[0].value, ';EndpointSuffix=',toLower(variables('StorageSuffix')))]",
+                        "logAnalyticsUri": "[variables('LogAnaltyicsUri')]",
+                        "WorkspaceID": "[parameters('WorkspaceID')]",
+                        "WorkspaceKey": "[parameters('WorkspaceKey')]",
+                        "SentinelOneUser": "[parameters('SentinelOneUser')]",
+                        "SentinelOnePassword": "[parameters('SentinelOnePassword')]",
+                        "SentinelOneUrl": "[parameters('SentinelOneUrl')]",
+                        "WEBSITE_RUN_FROM_PACKAGE": "https://github.com/averbn/azure_sentinel_data_connectors/blob/main/sentinelone-azure-sentinel-data-connector/SentinelOneAPISentinelConn.zip?raw=true"
```
Comment (v-admahe) [inline DataConnectors/SentinelOne/azuredeploy_Connector_SentinelOneAPI_AzureFunction.json:162]: Please replace value of WEBSITE_RUN_FROM_PACKAGE with https://aka.ms/sentinel-SentinelOneAPI-functionapp

Thanks,
Aditi Maheshwari

**Review by v-admahe — CHANGES_REQUESTED** [review]

**Review by v-rucdu — APPROVED** [review]

**Review by v-jayakal — APPROVED** [review]

**Pull request merged**